addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_nodes.py

```diff
@@ -114,9 +116,47 @@ def __gather_children(blender_object, export_settings):
 
     # blender bones
     if blender_object.type == "ARMATURE":
+        root_joints = []
         for blender_bone in blender_object.pose.bones:
             if not blender_bone.parent:
-                children.append(gltf2_blender_gather_joints.gather_joint(blender_bone, export_settings))
+                joint = gltf2_blender_gather_joints.gather_joint(blender_bone, export_settings)
+                children.append(joint)
+                root_joints.append(joint)
+        # handle objects directly parented to bones
+        direct_bone_children = [child for child in blender_object.children if child.parent_bone]
+        def find_parent_joint(joints, name):
+            for joint in joints:
+                if joint.name == name:
+                    return joint
+                parent_joint = find_parent_joint(joint.children, name)
+                if parent_joint:
+                    return parent_joint
+            return None
+        for child in direct_bone_children:
+            # find parent joint
+            parent_joint = find_parent_joint(root_joints, child.parent_bone)
+            if parent_joint:
+                child_node = gather_node(child, export_settings)
+                if child_node is not None:
```

Comment:
nit: could you change the `if parent_joint:` and `if child_node is not None:` checks to use a "guard clause" pattern, avoiding the extra nesting? For example:

```python
if not parent_joint:
  continue
# ...
```

Reply:
done